pkgs/test/nixpkgs-check-by-name/scripts/run-local.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2016
+
+set -euo pipefail
+
+cleanup_commands=()
+cleanup() {
+    echo -n >&2 "Cleaning up.. "
+    # Run all cleanup commands in inverse order
+    for (( i=${#cleanup_commands[@]}-1; i>=0; i-- )); do
+        eval "${cleanup_commands[i]}"
```

Comment:
This poses a lot of danger. Are there actually any commands which need to be `eval`ed here? The ones below all seem to refer to variables which are already in scope, so we could just do

```
cleanup_commands+=("git worktree remove --force '$tmp/merged'")
```

and then

```
"${cleanup_commands[i]}"
```

in this line.

Reply:
That doesn't work because it would try to execute `git worktree remove --force '$tmp/merged'` as a file (just tried it out).

Reply:
My bad, nested quotes always get me. Would it be possible to write the cleanup commands to a temporary file? That should be safer than `eval`.

Reply:
Are you thinking of something like this?
```bash
touch foo
echo "rm foo" >> "$tmp/cleanup_commands"

source "$tmp/cleanup_commands"
```

That sounds just like the `eval` approach, just with one more level of indirection.

---

Is there really a safety risk with `eval` here? It seems fine to me.

Reply:
`eval` is the biggest footgun in Bash. If anyone changes any of the cleanup commands to anything even slightly more complex in the future, the potential for disaster is there.

Reply:
We discussed this [on Matrix](https://matrix.to/#/!kjdutkOsheZdjqYmqp:nixos.org/$XmMLYPzkkX2ddFYTxGBTrZBDm7Yz92ggmK_IXGOLRnY?via=nixos.org&via=matrix.org&via=nixos.dev). I updated the script accordingly now: b62b605c23b6ff7771ac8bf9a2c78ee2ddd69b14